### PR TITLE
FeeHandler auditors comments

### PR DESCRIPTION
**Pull request by pahor167**
### Description

Addressing following audit comments

CLAB-25. INACTIVE TOKENS CAN STILL BE PROCESSED IN FEEHANDLER - updated in a way that removed tokens cannot be processed
CLAB-26. WRONG CALCULATION FOR TOKEN DISTRIBUTION FOR SELL
CLAB-29. INCORRECT CALCULATION OF BURN FRACTION AND DISTRIBUTION AMOUNT
CLAB-28. INCORRECT MINIMAL LIMIT OF BURNING TOKEN
CLAB-1. DUPLICATE ARRAY LENGTH CHECK IN FEEHANDLER
CLAB-12. AMBIGUOUS LOGIC FOR MAX SLIPPAGE IN FEEHANDLERSELLER
CLAB-17. REDUNDANT LINE AND HANDLER CHECK IN FEEHANDLER
CLAB-18. UNUSED EVENTS

### Other changes

No other changes

### Tested

Unit tests updated and added

### CODE REVIEW

File: packages/protocol/contracts/common/FeeHandler.sol [hunk 1]
```diff
@@ -234,13 +231,11 @@ contract FeeHandler is
...
 
   function _addToken(address tokenAddress, address handlerAddress) private {
-    // Check that the contract implements the interface
-    IFeeHandlerSeller(handlerAddress);
-
     TokenState storage tokenState = tokenStates[tokenAddress];
     tokenState.handler = handlerAddress;
 
     activeTokens.add(tokenAddress);
+    emit TokenAdded(tokenAddress, handlerAddress)
```
Comment (soloseng) [inline packages/protocol/contracts/common/FeeHandler.sol]:
```suggestion
    emit TokenAdded(tokenAddress, handlerAddress);
```

File: packages/protocol/contracts/common/FeeHandler.sol [hunk 2]
```diff
@@ -448,6 +451,7 @@ contract FeeHandler is
   }
 
   function handle(address tokenAddress) external {
```
Comment (soloseng) [inline packages/protocol/contracts/common/FeeHandler.sol:467]: add NATSPEC here and other external functions without.

**Review by soloseng — APPROVED** [review]
Looks good! 🚀

File: packages/protocol/lib/test-utils.ts [hunk 3]
```diff
@@ -637,3 +637,20 @@ export const unlockAndAuthorizeKey = async (offset: number, authorizeFn: any, ac
     from: account,
   })
 }
+
+export function expectBigNumberInRange(real: BigNumber, expected: BigNumber, range: BigNumber) {
+  console.log("real", real.toString());
+  console.log("expected", expected.toString());
```
Comment (montera82) [inline packages/protocol/lib/test-utils.ts]: do we need these?

File: packages/protocol/contracts/common/FeeHandler.sol [hunk 4]
```diff
@@ -348,6 +356,11 @@ contract FeeHandler is
     uint256 tokenBalance = token.balanceOf(address(this));
 
     TokenState storage tokenState = tokenStates[tokenAddress];
+    require(
```
Comment (martinvol) [inline packages/protocol/contracts/common/FeeHandler.sol:359]: Is this require needed? tokens can be distributed even if there's not handler set.

Comment (pahor167) [reply packages/protocol/contracts/common/FeeHandler.sol:359]: As discussed offline - this is there to be able to stop handling removed tokens altogether

**Pull request merged**